xonsh/pyghooks.py

```diff
@@ -281,7 +282,7 @@ def __init__(self, style_name='default'):
         style_name : str, optional
             The style name to initialize with.
         """
-        self.trap = {}  # for custom colors
+        self.trap = builtins.__xonsh_env__.get('PROMPT_TOOLKIT_STYLES') or {}
```

Comment:
I am not such a huge fan of this.  I think that the trap serves basically the same purpose as setting the user colors. Also, this has less to do really with prompt-toolkit and more to do with pygments.  I would be in favor of totally deprecating the `$PROMPT_TOOLKIT_STYLES` variable and replacing it with a `$XONSH_USER_COLORS` or `$XONSH_COLOR_MASK` or similar.  This could then be used across both pygments and ansicolors as needed. 

The rest of the code looks ok.

Does this sound reasonable?